docs/ibc/light-clients/client-state.md

```diff
@@ -1,3 +1,65 @@
 <!--
 order: 2
--->
\ No newline at end of file
+-->
+
+# Implementing the ClientState interface
+
+Learn how to implement the [Client State](https://github.com/cosmos/ibc-go/blob/main/modules/core/exported/client.go#L36) interface.
+
+### ClientType
+
+`ClientType` should return a unique string identifier of the light client.
+
+### GetLatestHeight
+
+`GetLatestHeight` should return the latest block height.
+
+### Validate
+
+`Validate` should validate every client state field and should return an error if any value is invalid. The light client
+implementor is in charge of determining which checks are required. See the [tendermint light client implementation](https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/client_state.go#L111)
+as a reference.
+
+### Status
+
+`Status` must return the status of the client. Only `Active` clients are allowed to process packets. All
+possible Status types can be found [here](https://github.com/cosmos/ibc-go/blob/main/modules/core/exported/client.go).
+
+### ZeroCustomFields
+
+`ZeroCustomFields` should return a copy of the light client with all client customizable fields with their zero value. It should not mutate the fields of the light client.
+This method is used to [verify upgrades](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/proposal.go#L120) and when [scheduling upgrades](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/proposal.go#L82).
```

Comment:
Maybe we can link to documentation on what client upgrades are? We could also add a short note that they are used to upgrade chain specific fields such as chainID, unbonding period. That is, when a chain decides to change its name or some other chain specific field, our light client must go through a specific process to be updated, otherwise regular client updates will break when the chain undergoes an upgrade